Notifiable-iOS/FWTNotifiableManager.m

```diff
@@ -74,25 +74,34 @@ - (void)dealloc
 
 + (NSHashTable *)listenerTable
 {
-    if (listeners == nil) {
-        listeners = [NSHashTable weakObjectsHashTable];
+    __block NSHashTable* table;
+    @synchronized(self) {
+        if (listeners == nil) {
+            listeners = [NSHashTable weakObjectsHashTable];
+        }
+        table = listeners;
     }
-    return listeners;
+    return table;
 }
 
 + (NSHashTable *)managerListenerTable
 {
-    if (managerListeners == nil) {
-        managerListeners = [NSHashTable weakObjectsHashTable];
+    __block NSHashTable* table;
+    @synchronized(self) {
+        if (managerListeners == nil) {
+            managerListeners = [NSHashTable weakObjectsHashTable];
+        }
+        table = managerListeners;
     }
-    return managerListeners;
+    return table;
 }
 
 + (void) operateOnListenerTableOnBackground:(void(^)(NSHashTable *table, NSHashTable *managerTable))block
 {
+    __block NSHashTable *table = [FWTNotifiableManager listenerTable];
```

Comment:
`__block` qualifier is unneeded since the block doesn't update the variable itself